internal/service/currency_service.go

```diff
@@ -88,49 +77,44 @@ func getCurrencyCode(currencyCode string) int {
 	if err != nil {
 		log.Fatal("Currency code is not a number")
 	}
-
 	return numericCode
 }
-
 func (s *CurrencyService) getCachedRate() (float64, bool) {
 	s.cacheMutex.Lock()
 	defer s.cacheMutex.Unlock()
-
 	if time.Since(s.cacheTime) < time.Hour {
 		return s.cacheRate, true
 	}
 	return 0, false
 }
-
 func fetchRateFromAPI() (float64, error) {
```

Comment:
this is not related to CurrencyService i think it would be better to move this to util